listenbrainz/webserver/static/js/src/stats/UserListeningActivity.tsx

```diff
@@ -0,0 +1,455 @@
+import * as React from "react";
+import MediaQuery from "react-responsive";
+
+import APIService from "../APIService";
+import Card from "../components/Card";
+import BarDualTone from "./BarDualTone";
+import Loader from "../components/Loader";
+
+export type UserListeningActivityProps = {
+  range: UserStatsAPIRange;
+  user: ListenBrainzUser;
+  apiUrl: string;
+};
+
+export type UserListeningActivityState = {
+  data: UserListeningActivityData;
+  lastRangePeriod: {
+    start?: Date;
+    end?: Date;
+  };
+  thisRangePeriod: {
+    start?: Date;
+    end?: Date;
+  };
+  loading: boolean;
+  totalListens: number;
+  avgListens: number;
+};
+
+export default class UserListeningActivity extends React.Component<
+  UserListeningActivityProps,
+  UserListeningActivityState
+> {
+  APIService: APIService;
+
+  rangeMap = {
```

Comment:
Type for this?

Reply:
The type is automatically inferred from the object, we don't need to provide it explicitly.